lib/order_repository.rb

```diff
@@ -0,0 +1,84 @@
+require_relative './order'
+class OrderRepository
+
+    # Selecting all records
+    # No arguments
+    def all
+      # Executes the SQL query:
+      # SELECT id, customer_name, date FROM orders;
+      sql = 'SELECT id, customer_name, date FROM orders;'
+      result_set = DatabaseConnection.exec_params(sql, [])
+
+      orders = []
+
+      result_set.each do |record|
+        order = Order.new
+        order.id = record['id']
+        order.customer_name = record['customer_name']
+        order.date = record['date']
+
+        orders << order
+      end
+      return orders
+      # Returns an array of Order objects.
+    end
+  
+    # Gets a single record by its ID
+    # One argument: the id (number)
+    def find(id)
+      # Executes the SQL query:
+      # SELECT id, customer_name, date FROM orders WHERE id = $1;
+      sql = 'SELECT id, customer_name, date FROM orders WHERE id = $1;'
+      sql_params = [id]
+
+      result_set = DatabaseConnection.exec_params(sql, sql_params)
+
+      record = result_set[0]
+
+      order = Order.new
```

Comment:
You could turn this chunk of code into it's own private method as it is repeated a couple of times